crates/bevy_window/src/window.rs

```diff
@@ -621,20 +619,18 @@ pub struct WindowResolution {
 impl Default for WindowResolution {
     fn default() -> Self {
         WindowResolution {
-            physical_width: 1280,
-            physical_height: 720,
+            physical_size: PhysicalSize::new(1280, 720),
             scale_factor_override: None,
             scale_factor: 1.0,
         }
     }
 }
 
 impl WindowResolution {
-    /// Creates a new [`WindowResolution`].
-    pub fn new(logical_width: f32, logical_height: f32) -> Self {
+    /// Creates a new [`WindowResolution`] with a scale factor of 1.0.
```

Comment:
This comment is wrong I think, the new function takes a logical size, and stores it in the physical size field. Then the windowing backend or whatever determines the scale factor and multiplies the logical size stored in the physical size by the scale factor to get the final physical size value.

Reply:
I agree that it is technically correct, as creating the struct by itself without the window will have a scale of 1.0. But I also agree, it's really not that helpful and misleading to say the scale is 1.0 if it will be changed later for most uses of it. For `Window`, the resolution scale factor looks to be set by the OS display scale, at least on Linux.

So, a change to address this might be: add "logical" back to naming, keep the `u32`s so there is no casting, change the doc to specify that winit determines the scale factor of the `WindowResolution`.
```rust
/// Creates a new [`WindowResolution`] with a scale factor of 1.0.
///
/// The [`WindowResolution`] scale factor in [`Window`] will be determined by winit.
    pub fn new(logical_width: u32, logical_height: u32) -> Self {
        Self {
            physical_size: PhysicalSize::new(width, height),
            ..Default::default()
        }
    }
```
Only saying that the scale factor is set by winit would be "technically" incorrect as the struct used alone will have the default of 1.0, but only when a part of `Window` is its scale factor set by winit.

It might also be worth describing the part where the logical size is stored in `physical_size` before getting the actual physical size?

Reply:
> Only saying that the scale factor is set by winit would be "technically" incorrect as the struct used alone will have the default of 1.0, but only when a part of `Window` is its scale factor set by winit.
> 

Mentioning `winit` doesn't seem right either, it's what most people use but Bevy is meant to backend agnostic.

Reply:
I've been thinking that `Window::Resolution` needs to be refactored. I really dislike this design where you request a logical resolution by setting a physical size.

My prefered API I think would be something like this:
```rust
pub struct Window {
    /// Desired resolution for the window
    pub target_resolution: TargetWindowResolution,
    /// Resolution of the window, automatically set on creation
    pub current_resolution: Option<WindowResolution>,
    // .. rest of Window's fields
}

pub struct TargetWindowResolution {
    pub size: LogicalSize,
    pub scale_factor_override: Option<f32>, 
}

pub struct CurrentWindowResolution {
    pub size: PhysicalSize,
    pub scale_factor: f32,
}
```

We could also use an enum, but I don't like how you lose the target size on window creation:
```rust
pub enum WindowResolution {
    Desired {
        size: LogicalSize,
        scale_factor_override: Option<f32>,              
    },
    Realized {
        size: PhysicalSize,
        scale_factor: f32,
        scale_factor_override: Option<f32>,
    }
}
```

That needs to be a separate PR though, this one has a lot of changes already.

Reply:
> Mentioning `winit` doesn't seem right either, it's what most people use but Bevy is meant to backend agnostic.

What about changing "winit" to "window backend" then it would be : 
```rust
...
/// The [`WindowResolution`] scale factor in [`Window`] will be determined by the window backend.
```